### DISCUSSION TITLE
Websockets with com.c4-soft.springaddons:spring-addons-webflux-jwt-resource-server

### DISCUSSION THREAD

**Discussion by virtex-br** [ch4mpy/spring-addons#224] [Q&A] [answered]
Hi,

I'm using the spring-addons-webflux-jwt-resource-server pack, which works brilliantly. 

I'm configuring a Websocket using:
`registry.addHandler(new` WebSocketHandler(), "/websocket").setAllowedOrigins("*");`
which seems to be disabled somehow by the plugin.

Can you advise the best way to hook up a simple websocket endpoint while using the addon?

Thanks!

**Comment by ch4mpy** [comment, marked as answer]
Starting from version `7.8.7`, CORS configuration is done with a global filter. Usage and default configuration overrides are described in [the release notes](https://github.com/ch4mpy/spring-addons/blob/master/release-notes.md#787). I just fixed [the `spring-addons-starter-oidc` README](https://github.com/ch4mpy/spring-addons/blob/master/spring-addons-starter-oidc/README.MD#116-cors-configuration) to document the new behavior.

### P.S.
At the age of virtual threads, I'd rather start from a servlet sample (the `webmvc` variants), because of the so much easier code debugging. I only use WebFlux when working with Spring Cloud Gateway - and when maintaining these samples.

Also, I carefully avoid cross origins situations in my applications because requests from the frontends are authorized with `SameSite` cookies. **I strongly advise that you take the time to read [this article about the OAuth2 BFF pattern](https://www.baeldung.com/spring-cloud-gateway-bff-oauth2).**

**Reply by virtex-br** [reply]
Superb - thanks for the speedy response. Another question - is there an easy way to configure the system to use JWT and X-API-TOKEN headers at the same time?

**Reply by ch4mpy** [reply]
New question, [new thread](https://github.com/ch4mpy/spring-addons/discussions/225).
